Microsoft.Azure.CosmosRepository/src/Providers/DefaultCosmosClientProvider.cs

```diff
@@ -13,6 +13,7 @@ namespace Microsoft.Azure.CosmosRepository.Providers
     class DefaultCosmosClientProvider : ICosmosClientProvider, IDisposable
     {
         readonly Lazy<CosmosClient> _lazyCosmosClient;
+        readonly ICosmosClientOptionsProvider _cosmosClientOptionsProvider;
```

Comment:
Isn't this is a breaking change? If the consumer the `.AddCosmosRepository(IConfiguration config)` overload (which all users are currently using now), this will break them as this is only added in the Azure function path.

Might be easier to just register your own `CosmosClientOptions` instance externally. I'll have to think about this a bit, and try it out later.

Reply:
I'm not sure I understand. Are you suggesting to create a separate AddCosmosRepository() (named, for example, AddCosmosRepositoryForFunction()) ?
As for externally registering CosmosClientOptions, it is possible to register the RepositoryOptions setting section then extract the IConfiguration instance but it breaks the dependency injection pattern. That's why I did this feature.

In fact, I wanted to make this available for all .NET Core project type but, I'm not sure that's possible due to different ways to store and load settings. Or may be there's a way to ConnectionString, DatabaseId and other settings in the same IConfiguration object but I don't know the ServiceTier and WebTier app type enough.

Reply:
Hi @IEvangelist,
Did you have time to think about it ?
If you want, we could talk about it over Teams (now that we all are used to it !), it will be much quickier for you to explain it to me. Anyway, I'm available any time :)

Reply:
Sorry for the delay, yes - let's jump on a call quick. Email me, my email is on my GitHub profile @IEvangelist.

Reply:
I just sent you an email :)